### PR TITLE
Allow to add some extra text to created hmplugin file

### PR DESCRIPTION
**Pull request by prusse-martin**
ASIM-5082

(I will need to deploy this before running the test in alfasim-sdk)

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -2,11 +2,17 @@
 History
 =======
 
+0.5.0 (2023-??-??)
```
Comment (nicoddemus) [inline HISTORY.rst:5]: We usually do this:

```suggestion
0.5.0 (UNRELEASED)
```

File: hookman/hookman_generator.py [hunk 2]
```diff
@@ -284,6 +285,9 @@ def generate_plugin_package(
 
         :param Dict[str,str] extras_defaults:
             (key, value) entries to be added to "extras" if not defined by the original input yaml.
+
+        :param Optional[str] package_name_extra:
```
Comment (nicoddemus) [inline hookman/hookman_generator.py:289]:
```suggestion
        :param Optional[str] package_name_suffix:
```

File: hookman/hookman_generator.py [hunk 3]
```diff
@@ -314,6 +311,18 @@ def generate_plugin_package(
             contents_dict["extras"] = dict(sorted(extras.items()))
             contents = contents_dict.as_yaml()
 
+        hmplugin_base_name_components = [package_name, plugin_info.version]
+        if package_name_extra is not None:
+            hmplugin_base_name_components.append(package_name_extra)
+
+        if sys.platform == "win32":  # pragma: no cover (apparently merge reports fo the same files)
```
Comment (nicoddemus) [inline hookman/hookman_generator.py:318]:
```suggestion
        if sys.platform == "win32":  # pragma: no cover (apparently merge reports of the same files)
```

**Review by cauebs — APPROVED** [review]

File: HISTORY.rst (hunk 1 above)
Comment (prusse-martin) [reply HISTORY.rst:5]: OK.
I was planning to make a release right after merging this (I would update that before merging).

File: hookman/hookman_generator.py (hunk 2 above)
Comment (prusse-martin) [reply hookman/hookman_generator.py:289]: I was originally planning to use `package_name_suffix` =P

- [ ] name `package_name_extra` as `package_name_suffix`;

File: hookman/hookman_generator.py (hunk 3 above)
Comment (prusse-martin) [reply hookman/hookman_generator.py]: foooooooo!

**Review by nicoddemus — APPROVED** [review]

**Review by BeneBr — APPROVED** [review]

**Pull request merged**